pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamConsumerFactory.java

```diff
@@ -19,12 +19,15 @@
 package org.apache.pinot.spi.stream;
 
 import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
 
 
 /**
  * Factory for a stream which provides a consumer and a metadata provider for the stream
  */
 public abstract class StreamConsumerFactory {
+  public static final AtomicInteger CLIENT_ID_SEQ = new AtomicInteger(0);
```

Comment:
Let's add a method `String getUniqueClientId(String prefix)` to access this to avoid the mis-use (e.g. currently there are mixed usage of `incrementAndGet()` and `getAndIncrement()`, which is a bug)

Reply:
done